rct229/rulesets/ashrae9012019/section5/section5rule43.py

```diff
@@ -0,0 +1,120 @@
+                list_path="$.building_segments[*].zones[*].surfaces[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building_p = context.proposed
+            return {
+                "scc_dict_p": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building_p
+                ),
+            }
+
+        def list_filter(self, context_item, data=None):
+            surface_p = context_item.proposed
+            return get_opaque_surface_type(surface_p) == OST.ROOF
+
+        class RoofRule(RuleDefinitionBase):
+            def __init__(self):
+                super(Section5Rule43.BuildingRule.RoofRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(True, False, True),
+                    required_fields={
+                        "$": ["surface_optical_properties"],
+                        "surface_optical_properties": ["absorptance_solar_exterior"],
+                    },
+                )
+
+            def get_calc_vals(self, context, data=None):
+                roof_p = context.proposed
+                roof_u = context.user
+                surface_conditioning_category_p = data["scc_dict_p"]
+
+                return {
+                    "absorptance_solar_exterior_p": roof_p[
+                        "surface_optical_properties"
+                    ]["absorptance_solar_exterior"],
+                    "absorptance_solar_exterior_u": roof_u[
+                        "surface_optical_properties"
+                    ]["absorptance_solar_exterior"],
+                    "surface_conditioning_category_p": surface_conditioning_category_p[
+                        roof_p["id"]
+                    ],
+                }
+
+            def rule_check(self, context, calc_vals=None, data=None):
+                return (
+                    calc_vals["absorptance_solar_exterior_p"]
+                    == calc_vals["absorptance_solar_exterior_u"]
+                    or calc_vals["absorptance_solar_exterior_p"]
+                    == ABSORPTION_SOLAR_EXTERIOR
+                )
+
+            def get_pass_msg(self, context, calc_vals=None, data=None):
+                """Pre-condition: see rule_check"""
+                absorptance_thermal_exterior_p = calc_vals[
+                    "absorptance_thermal_exterior_p"
+                ]
+                absorptance_thermal_exterior_u = calc_vals[
+                    "absorptance_thermal_exterior_u"
+                ]
+                surface_conditioning_category_p = calc_vals[
+                    "surface_conditioning_category_p"
+                ]
+                pass_msg = ""
+                if absorptance_thermal_exterior_p != ABSORPTION_SOLAR_EXTERIOR:
+                    # this condition only applies when P-RMD = U-RMD
+                    pass_msg = PASS_NOT_EQUAL_MSG
+                if absorptance_thermal_exterior_u != absorptance_thermal_exterior_p:
```

Comment:
I think this second `if` needs to be `elif`. And I think it could be further simplified to:
```
elif absorptance_thermal_exterior_u != absorptance_thermal_exterior_p and surface_conditioning_category_p != SCC.UNREGULATED:
    # this condition only applies when P-RMD = 0.7 and P-RMD surface is regulated
    pass_msg = PASS_DIFFERS_MSG_REGULATED
```

Whew! This logic was hard to check.

Reply:
I think you are correct. if should be elif. Thanks for checking!